lib/fluent/plugin/buffer/chunk.rb

```diff
@@ -149,11 +152,31 @@ def open(&block)
           raise NotImplementedError, "Implement this method in child class"
         end
 
-        def write_to(io)
+        def write_to(io, options = {})
           open do |i|
             IO.copy_stream(i, io)
           end
         end
+
+        module Decompressable
+          include Fluent::Plugin::Compressable
+
+          def write_to(io, options = {})
+            unless options[:compress] == :gzip
+              c = decompress(read)
+              if @chunk.is_a?(String)
+                @chunk = c
```

Comment:
Is it the hack for `MemoryChunk#write_to`, right?
IMO the implementation of `MemoryChunk#write_to` is wrong, and it should be `io.write self.read`.
If `self.read` returns decompressed value, there are no need to do such hack.